core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -198,7 +198,8 @@ public InternalEngine(EngineConfig engineConfig) {
                         seqNoStats = new SeqNoStats(
                                 SequenceNumbers.NO_OPS_PERFORMED,
                                 SequenceNumbers.NO_OPS_PERFORMED,
-                                SequenceNumbers.UNASSIGNED_SEQ_NO);
+                                SequenceNumbers.UNASSIGNED_SEQ_NO,
+                                getHistoryUUID());
```

Comment:
I am not comfortable with this invocation when history UUID has not been initialized yet.